src/client.js

```diff
@@ -95,4 +130,55 @@ export default class Client {
   on(event, cb) {
     this.eventEmitter.on(event, cb);
   }
+
+  /* ---- HELPERS ---- */
+  get(path, options) {
+    const _options = options || {};
+    _options.method = 'get';
+    return this.fetch(path, _options);
+  }
+
+  post(path, body, options) {
+    let _options = { ...this.defaults.post };
+    _options.body = JSON.stringify(body);
+    _options = merge.recursive(true, _options, options);
+    _options.method = 'post';
+    return this.fetch(path, _options);
+  }
+
+  put(path, body, options) {
+    let _options = { ...this.defaults.put };
+    _options.body = JSON.stringify(body);
+    _options = merge.recursive(true, _options, options);
+    _options.method = 'put';
+    return this.fetch(path, _options);
+  }
+
+  patch(path, body, options) {
+    let _options = { ...this.defaults.patch };
+    _options.body = JSON.stringify(body);
+    _options = merge.recursive(true, _options, options);
+    _options.method = 'patch';
+    return this.fetch(path, _options);
+  }
+
+  delete(path, options) {
+    const _options = options || {};
+    _options.method = 'delete';
+    return this.fetch(path, _options);
+  }
+
+  upload(path, input, options) {
+    const _options = options || {};
+    _options.method = 'post';
+
+    const data = new FormData();
+    let i;
+    for (i = 0; i < input.files.length; i++) {
+      data.append('files[]', input.files[i]);
```

Comment:
This isn't defaulting to an array - it takes the input element object which has a `files` array. I can always take the name instead and grab it from the DOM myself, but I kinda liked letting the app decide how it wants to interact with the DOM (even though this is read only and will have no side effects). Thoughts?

You're right, this should handle PUT/PATCH also.

You think the upload helper should take a object of other things to be appended into the body of the request instead of just uploading files? I suppose this could be redone as more of a "form data" function instead of just an "upload" function that takes an object, but then also needs to take an array of file input elements, right? I don't mind having a function like that, but it could exist side by side with this upload function. This can be for simply uploading from a single file input, which is a typical use case in my projects, and the "form data" function can handle a whole form, including multiple file inputs, and post/put/patch with the form-data header instead of as json.